### PR TITLE
docs(cognito): README touch-ups

### PR DESCRIPTION
**Pull request by shivlaks**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-cognito/README.md [hunk 1]
```diff
@@ -343,6 +352,36 @@ The following table lists the set of triggers available, and their corresponding
 For more information on the function of these triggers and how to configure them, read [User Pool Workflows with
 Triggers](https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-identity-pools-working-with-aws-lambda-triggers.html).
 
+You may need to provide your Lambda trigger with permissions to perform operations against your user pool.
```
Comment (nija-at) [inline packages/@aws-cdk/aws-cognito/README.md:355]: Create a subsection here called 'Trigger permissions' and starting off with a sentence on how to provide permission to lambda trigger. Then follow it up with a paragraph about the lambda trigger needing user pool permissions.

File: packages/@aws-cdk/aws-cognito/README.md [hunk 2]
```diff
@@ -343,6 +352,36 @@ The following table lists the set of triggers available, and their corresponding
...
+fn.role?.attachInlinePolicy(new Policy(stack, 'userpool-policy', {
+  statements: [new PolicyStatement({
+    actions: ['cognito-idp:DescribeUserPool'],
+    resources: [userpool.userPoolArn],
+  })],
+}));
+```
+
+⚠️ Using the `attachToRolePolicy` API to provide permissions to a Lambda trigger will result in a circular dependency error:
+> Circular dependency between resources: [pool056F3F7E, fnPostAuthFnCognitoA630A2B1, ...]
```
Comment (nija-at) [inline packages/@aws-cdk/aws-cognito/README.md:383]: Would be nice to state this before providing the work around.

**Review by nija-at — CHANGES_REQUESTED** [review]

File: packages/@aws-cdk/aws-cognito/README.md [hunk 3]
```diff
@@ -131,6 +135,8 @@ used additionally; or it can be configured so that email and/or phone numbers ar
 sign in. Read more about this
 [here](https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-settings-attributes.html#user-pool-settings-aliases-settings).
 
+⚠️ Username and Alias attributes `phone`, `email`, or `preferredUserName` are user pool properties that cannot be updated.
```
[the remark this answers is not in the document]
Comment (shivlaks) [reply packages/@aws-cdk/aws-cognito/README.md]: done

File: packages/@aws-cdk/aws-cognito/README.md (hunk 1 above)
Comment (shivlaks) [reply packages/@aws-cdk/aws-cognito/README.md]: took a run at it!

File: packages/@aws-cdk/aws-cognito/README.md (hunk 2 above)
Comment (shivlaks) [reply packages/@aws-cdk/aws-cognito/README.md]: agreed. moved this up to highlight it before offering the workaround.

File: packages/@aws-cdk/aws-cognito/README.md [hunk 4]
```diff
@@ -345,6 +355,40 @@ The following table lists the set of triggers available, and their corresponding
 For more information on the function of these triggers and how to configure them, read [User Pool Workflows with
 Triggers](https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-identity-pools-working-with-aws-lambda-triggers.html).
 
+#### Trigger Permissions
+
+You can provide permissions to your Lambda trigger using the `attachToRolePolicy` API which will update
+the execution role used by the Lambda function.
```
Comment (nija-at) [inline packages/@aws-cdk/aws-cognito/README.md:361]: The language throughout this doc avoids the 2nd person.

```suggestion
The `function.attachToRolePolicy()` API can be used to add additional IAM permissions to the lambda trigger
as necessary.
```

File: packages/@aws-cdk/aws-cognito/README.md [hunk 5]
```diff
@@ -345,6 +355,40 @@ The following table lists the set of triggers available, and their corresponding
...
+
+⚠️ Using the `attachToRolePolicy` API to provide permissions scoped to your user pool will result in a circular dependency.
```
Comment (nija-at) [inline packages/@aws-cdk/aws-cognito/README.md:363]:
```suggestion
⚠️ Using the `attachToRolePolicy` API to provide permissions to your user pool will result in a circular dependency. See [aws/aws-cdk#7016](https://github.com/aws/aws-cdk/issues/7016).
```

File: packages/@aws-cdk/aws-cognito/README.md [hunk 6]
```diff
@@ -345,6 +355,40 @@ The following table lists the set of triggers available, and their corresponding
...
+Error message when running `cdk synth` or `cdk deploy`:
+> Circular dependency between resources: [pool056F3F7E, fnPostAuthFnCognitoA630A2B1, ...]
+
+To work around the circular dependency issue, your Lambda trigger can be granted permissions to perform operations
+on your user pool using the `attachInlinePolicy` API.
```
Comment (nija-at) [inline packages/@aws-cdk/aws-cognito/README.md:368]:
```suggestion
To work around the circular dependency issue, use the `attachInlinePolicy()` API instead, as shown below.
```

File: packages/@aws-cdk/aws-cognito/README.md [hunk 7]
```diff
@@ -345,6 +355,40 @@ The following table lists the set of triggers available, and their corresponding
...
+});
+
+// provide permissions to describe the user pool scoped to the ARN of 'myuserpool'
+postAuthFn.role?.attachInlinePolicy(new iam.Policy(this, 'userpool-policy', {
+  statements: [new iam.PolicyStatement({
+    actions: ['cognito-idp:DescribeUserPool'],
+    resources: [userpool.userPoolArn],
+  })],
+}));
+```
```
Comment (nija-at) [inline packages/@aws-cdk/aws-cognito/README.md:390]: This is 15 lines of code with only the last 5 lines of actual value. Anything we can do to reduce the amount of boilerplate?

Not a major comment.

**Review by nija-at — APPROVED** [review]

File: packages/@aws-cdk/aws-cognito/README.md (hunk 7 above)
Comment (shivlaks) [reply packages/@aws-cdk/aws-cognito/README.md:390]: I could create a new fixture specifically for this snippet. would you prefer that here? - it's primarily this long for rosetta compatibility

Comment (nija-at) [reply packages/@aws-cdk/aws-cognito/README.md:389]: Either that or linking to the specific issue comment that has this workaround?

Comment (shivlaks) [reply packages/@aws-cdk/aws-cognito/README.md]: created a fixture (didn't quite read your comment). we can also link to comment if it's helpful, but i figured at least the fixture will help users in other languages as well.

**Pull request merged**